### ISSUE TITLE
Error when using mitmdump with script with print() statements

### ISSUE THREAD

**Issue by ujjwal96** [mitmproxy/mitmproxy#2216]
##### Steps to reproduce the problem:

1. Make a script with print statement
```
def response(flow):
    print("Test")
```
2. Run mitmdump. `mitmdump -s script.py`
3.
 ```
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addons/termlog.py", line 27, in log
    err=(e.level == "error")
  File "/home/ujjwal/Github/mitmproxy/venv/lib/python3.5/site-packages/click/termui.py", line 420, in secho
    return echo(style(text, **styles), file=file, nl=nl, err=err, color=color)
  File "/home/ujjwal/Github/mitmproxy/venv/lib/python3.5/site-packages/click/utils.py", line 259, in echo
    file.write(message)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addons/script.py", line 78, in write
    self.log(buf)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/log.py", line 31, in warn
    self(txt, "warn")
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/log.py", line 40, in __call__
    self.master.add_log(text, level)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/master.py", line 67, in add_log
    self.addons.trigger("log", log.LogEntry(e, level))
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addonmanager.py", line 155, in trigger
    return
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/master.py", line 49, in handlecontext
    yield
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addonmanager.py", line 153, in trigger
    self.invoke_addon(i, name, *args, **kwargs)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addonmanager.py", line 144, in invoke_addon
    func(*args, **kwargs)
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/addons/termlog.py", line 21, in log
    if self.options.verbosity >= log.log_tier(e.level):
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/optmanager.py", line 165, in __getattr__
    return self._options[attr].current()
  File "/home/ujjwal/Github/mitmproxy/mitmproxy/optmanager.py", line 51, in current
    v = self.default
RecursionError: maximum recursion depth exceeded
```

##### Any other comments? What have you tried so far?
It is probably caused by https://github.com/mitmproxy/mitmproxy/pull/2207

##### System information
Mitmproxy version: 3.0.0 (2.0.0dev0255-0x6f00c1b) 
Python version: 3.5.2
Platform: Linux-4.4.0-62-generic-x86_64-with-elementary-0.4-loki
SSL version: OpenSSL 1.0.2g  1 Mar 2016
Linux distro: "elementary" 0.4 loki

**Comment by nikofil** [comment]
You're right, it's because in mitmdump logs are printed again to stdout, causing an infinite recursion. I will fix this shortly.

**Comment by cortesi** [comment]
Fixed in #2261
